pkg/controller/backup_controller.go

```diff
@@ -230,6 +234,16 @@ func (b *backupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	switch original.Status.Phase {
 	case "", velerov1api.BackupPhaseNew:
 		// only process new backups
+	case velerov1api.BackupPhaseInProgress:
+		// if backup is in progress, we should not process it again
+		// we want to mark it as failed to avoid it being stuck in progress
+		// if this backupReconciler uuid is the same as the one in the backup, then we are the one who is processing it
+		// if so, mark it as failed, last loop did not successfully complete the backup
+		if original.Annotations[velerov1api.BackupControllerUID] == string(b.uuid) {
```

Comment:
UUID here is to address a comment above `switch original.Status.Phase {`
quoted below. We can discuss if it is necessary, or if we need a real leaderelection in this thread.
```
	// Double-check we have the correct phase. In the unlikely event that multiple controller
	// instances are running, it's possible for controller A to succeed in changing the phase to
	// InProgress, while controller B's attempt to patch the phase fails. When controller B
	// reprocesses the same backup, it will either show up as New (informer hasn't seen the update
	// yet) or as InProgress. In the former case, the patch attempt will fail again, until the
	// informer sees the update. In the latter case, after the informer has seen the update to
	// InProgress, we still need this check so we can return nil to indicate we've finished processing
	// this key (even though it was a no-op).
```

Reply:
From my team scrum this is not necessary, removing.